api/controllers/FileUploadController.js

```diff
@@ -61,11 +61,13 @@ const FileUploadController = {
         return multer(multerOptions).array(FORM_INPUT_NAME);
     },
 
-    _clearExistingErrorMessages(req) {
-        req.session.eApp.uploadMessages.errors = [];
-        req.session.eApp.uploadMessages.fileCountError = false;
-        req.session.eApp.uploadMessages.infectedFiles = [];
-        req.session.eApp.uploadMessages.noFileUploadedError = false;
+    clearExistingErrorMessages(req) {
+        req.session.eApp.uploadMessages = {
+            errors: [],
+            infectedFiles: [],
+            fileCountError: false,
+            noFileUploadedError: false,
+        }
     },
```

Comment:
instead of doing this manually, you probably actually want to use a feature like "flash" for this, sorry I didn't catch this before.

it's a fairly common server decoration/utility regardless of which framework you use (i.e. hapi/sails/express). You use it for showing short-lived messages. Typically, as soon as the message is read, it will delete itself or is deleted on the next request. 

It would work like this
1. user uploads erroneous file  (POST)
2. server detects error
3. server uses req.flash to store the error message(s)
4. response redirects user to the page with the error (REDIRECT)
5.  request is now GET. In the get request, if it can retrieve a message from req.flash, display it

Post-Redirect-Get is a common and well documented pattern, and using req.flash suits it quite well. 

unfortunately as of sails v1, they've removed the flash middleware and you have to install it yourself. https://sailsjs.com/documentation/upgrading/to-v-1-0#other-breaking-changes. they suggest this: https://www.npmjs.com/package/connect-flash